src/glue.rs

```diff
@@ -65,68 +73,140 @@ impl MockedCompiler {
     }
 }
 
+/// The compiler is providing us with a couple of [callbacks](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_driver/trait.Callbacks.html),
+/// that allow us to hook into its lifecycle
 impl Callbacks for MockedCompiler {
+    /// We are going to ask the compiler to build our previous and next crates,
+    /// instead of the regular lib.rs or main.rs.
     fn config(&mut self, config: &mut Config) {
         let (file_name, code) = self.file_name_and_code();
 
-        // Replace code with contained String
         config.input = Input::Str {
             name: FileName::Custom(file_name.to_owned()),
             input: code.to_owned(),
         }
     }
 
+    /// This is where everything is happening.
+    /// We will focus on [after_analysis](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_driver/trait.Callbacks.html#method.after_analysis),
+    /// where we will hook our breaking API checks.
     fn after_analysis<'tcx>(
         &mut self,
         _compiler: &rustc_interface::interface::Compiler,
         queries: &'tcx rustc_interface::Queries<'tcx>,
     ) -> Compilation {
-        let diff = queries
-            .global_ctxt()
-            .unwrap()
-            .take()
-            .enter(|tcx| get_diagnosis(&tcx));
+        // get prev & next
+        let changeset = queries.global_ctxt().unwrap().take().enter(|tcx| {
+            let comparator = ApiComparator::from_tcx(tcx)?;
+            get_changeset(comparator)
+        });
 
-        *self = MockedCompiler::Finished(diff);
+        *self = MockedCompiler::Finished(changeset);
 
+        // we don't need the compiler to actually generate any code.
         Compilation::Stop
     }
 }
 
-fn get_diagnosis(tcx: &TyCtxt) -> AnyResult<ApiCompatibilityDiagnostics> {
-    let (prev, curr) =
-        get_previous_and_current_nums(tcx).context("Failed to get dependencies crate id")?;
-    let comparator = ApiComparator::from_crate_nums(prev, curr, tcx);
+pub struct ChangeSet {
```

Comment:
the change set should not be defined in the glue file

(nothing should be defined in the glue file :sob:)